ppd/ppd-ipp.c

```diff
@@ -547,7 +547,7 @@ ppdLoadAttributes(
       while (*ptr && isspace(*ptr)) ptr ++;
       if (!isdigit(*ptr))
       {
-	strcpy(buf, ptr);
+	snprintf(buf, sizeof(buf), "%s", ptr);
 	ptr = buf;
```

Comment:
Note that, at least for the string in `buf` comming from a correct `*cupsFilter(2): ...` line, the crash cannot have been caused by the string pointed at by `ptr` being too long for `buf`, as `ptr` is pointing somewhere into `buf`. The problem is that, according to the man page, `strcpy()` is not suitable for copying overlapping strings/buffers, and so that could have caused the crash here.

Now, with the replacement by `snprintf()` the crash has gone away in this particular case, but here the man page also says that this function is not suitable for overlapping strings/buffers, so one should expect that a crash still can happen in other situations.

To be safe one could eaither do a `for()` loop moving the characters to the beginning of the buffer one-by-one or the `memmove()` function which according to its man page supports overlapping memory areas.

```
memmove(buf, ptr, strlen(ptr));
```
could be used here and should work reliably, at least if buf contains a correctly `\0`-terminated string.

Reply:
`strlen(ptr) + 1` (to also move the terminating nul character)